### PR TITLE
Update GameManager done

### PR DESCRIPTION
**Pull request by DullySkull**
Pause work, Game Over works
Singelton exist already and gives error need help tomarrow new scenes created for turtorial and main menu

### CODE REVIEW

File: Assets/_Project/Runtime/_Scripts/GameManager/GameManager.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        gameOverPanel.SetActive(false);
+        pausePanel.SetActive(false);
+    }
+
+    void Update()
+    {
+        switch (state)
+        {
+            case GameState.Play:
+                if (Input.GetKeyDown(KeyCode.Escape))
```
Comment (ltsLumina) [inline Assets/_Project/Runtime/_Scripts/GameManager/GameManager.cs:32]: I'll convert all of this to the new InputSystem eventually

File: Assets/_Project/Runtime/_Scripts/GameManager/TurtorialManager.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                // Player try recharge
+                break;
+            case GameState.TurtorialDone:
+                // Change scene
+                break;
+        }
+        // Add more cases if needed
+
+        GameStateChanger(state);
+    }
```
Comment (ltsLumina) [inline Assets/_Project/Runtime/_Scripts/GameManager/TurtorialManager.cs]: Love this. Good job :)

File: Assets/_Project/Runtime/_Scripts/UI Scripts/ScoreManager.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    [SerializeField] TextMeshProUGUI currentHolesRepaired;
+    [SerializeField] TextMeshProUGUI currentJellyfishCollected;
+    [SerializeField] TextMeshProUGUI currentLifeformsScanned;
+
+    [Header("Current Scores")]
+    public int depthScore;
+    public int algaeCollected;
+    public int holesRepaired;
+    public int jellyfishCollected;
+    public int lifeformsScanned;
```
Comment (ltsLumina) [inline Assets/_Project/Runtime/_Scripts/UI Scripts/ScoreManager.cs]: Don't make public variables. Use properties instead.
Example:

```c#
[SerializeField] int depthScore;

public int DepthScore => depthScore; // get only property

// or:

public int DepthScore
{
    get => depthScore;
    set => depthScore = value;
}
```

**Review by ltsLumina — CHANGES_REQUESTED** [review]
Looks good!
If you could just change the public variables to be separate properties that would be great :)
I'll take a look tomorrow again before we make the Alpha build!

File: Assets/_Project/Runtime/_Scripts/GameManager/GameManager.cs (hunk 1 above)
Comment (DullySkull) [reply Assets/_Project/Runtime/_Scripts/GameManager/GameManager.cs:32]: 👍

File: Assets/_Project/Runtime/_Scripts/GameManager/TurtorialManager.cs (hunk 2 above)
Comment (DullySkull) [reply Assets/_Project/Runtime/_Scripts/GameManager/TurtorialManager.cs]: 😄

File: Assets/_Project/Runtime/_Scripts/UI Scripts/ScoreManager.cs (hunk 3 above)
Comment (DullySkull) [reply Assets/_Project/Runtime/_Scripts/UI Scripts/ScoreManager.cs]: Fixed it! Did the same thing with high score

**Review by ltsLumina — APPROVED** [review]

**Pull request merged**